KeeTrayTOTP/TrayTOTP_Plugin.cs

```diff
@@ -200,18 +200,35 @@ internal void OnMenuHelpClick(object sender, EventArgs e)
         /// </summary>
         internal IEnumerable<PwEntry> GetVisibleAndValidPasswordEntries()
         {
-            return GetVisibleAndValidPasswordEntries(PluginHost.MainWindow.ActiveDatabase.RootGroup);
+            return GetVisibleAndValidPasswordEntries(PluginHost.MainWindow.ActiveDatabase);
         }
 
         /// <summary>
         /// Get all the password entries in all groups and filter entries that are expired or have invalid TOTP settings.
         /// </summary>
-        /// <returns></returns>
-        internal IEnumerable<PwEntry> GetVisibleAndValidPasswordEntries(PwGroup pwGroup)
+        internal IEnumerable<PwEntry> GetVisibleAndValidPasswordEntries(PwDatabase pwDatabase)
+        {
+            var entries = pwDatabase.RootGroup.GetEntries(true);
+
+            return entries.Where(entry => !entry.IsExpired() && TOTPEntryValidator.HasSeed(entry) && !InRecycleBin(pwDatabase, entry));
+        }
+
+        /// <summary>
+        /// Returns true when there is a recycle bin, and the entry is inside of it.
+        /// </summary>
+        /// <param name="entry"></param>
+        private bool InRecycleBin(PwDatabase pwDatabase, PwEntry entry)
         {
-            var entries = pwGroup.GetEntries(true);
+            if (pwDatabase.RecycleBinEnabled)
+            {
+                var pgRecycleBin = pwDatabase.RootGroup.FindGroup(pwDatabase.RecycleBinUuid, true);
```

Comment:
I did consider that, but did not look into it, because I added this check last in the function looking for totp entries, so it is indeed only called on items for which we know they are totp enabled. Pushed a little change that constructs a more optimal strategy which should look it up only once.